### PR TITLE
Remove std-lib reflect dependency

### PR DESCRIPTION
**Pull request by ayeshLK**

### CODE REVIEW

File: websubhub-ballerina/tests/additional_error_details_test.bal [hunk 1]
```diff
@@ -112,7 +112,7 @@ function testRegistrationFailureErrorDetails() returns @tainted error? {
     http:Request request = new;
     request.setTextPayload("hub.mode=register&hub.topic=test1", "application/x-www-form-urlencoded");
 
-    var response = check errorDetailsTestClientEp->post("/", request);
+    var response = errorDetailsTestClientEp->post("/", request);
```
Comment (ThisaruGuruge) [inline websubhub-ballerina/tests/additional_error_details_test.bal:115]: The ideal way (in tests) is to keep the `check` and remove the type guard.

File: websubhub-ballerina/tests/additional_error_details_test.bal [hunk 2]
```diff
@@ -113,20 +113,16 @@ function testRegistrationFailureErrorDetails() returns @tainted error? {
     request.setTextPayload("hub.mode=register&hub.topic=test1", "application/x-www-form-urlencoded");
 
     var response = check errorDetailsTestClientEp->post("/", request);
```
Comment (Bhashinee) [inline websubhub-ballerina/tests/additional_error_details_test.bal:115]:
```suggestion
    http:Response response = check errorDetailsTestClientEp->post("/", request);
```

File: websubhub-ballerina/tests/additional_error_details_test.bal [hunk 3]
```diff
@@ -138,19 +134,15 @@ function testDeregistrationFailureErrorDetails() returns @tainted error? {
...
-            test:assertEquals(responseBody["hub.mode"], "denied");
-            test:assertEquals(responseBody["hub.reason"], "Topic deregistration failed!");
-        }
+    test:assertEquals(response.statusCode, 200);
+    var payload = response.getTextPayload();
+    if (payload is error) {
+        test:assertFail("Could not retrieve response body for topic-deregistration failure");
     } else {
-        test:assertFail("Topic deregistration test failed");
+        var responseBody = decodeResponseBody(payload);
```
Comment (Bhashinee) [inline websubhub-ballerina/tests/additional_error_details_test.bal:142]:
```suggestion
        map<string> responseBody = decodeResponseBody(payload);
```
Check other places as well.

File: websubhub-ballerina/tests/additional_error_details_test.bal [hunk 4]
```diff
@@ -113,20 +113,16 @@ function testRegistrationFailureErrorDetails() returns @tainted error? {
...
-            test:assertFail("Could not retrieve response body for topic-registration failure");
-        } else {
-            var responseBody = decodeResponseBody(payload);
-            log:print("Retrieved error-response for topic registration failure ", responseBody = responseBody);
-            test:assertEquals(responseBody["hub.mode"], "denied");
-            test:assertEquals(responseBody["hub.reason"], "Topic registration failed!");
-            test:assertEquals(responseBody["hub.additional.details"], "Feature is not supported in the hub");
-        }
+    test:assertEquals(response.statusCode, 200);
+    var payload = response.getTextPayload();
```
Comment (ThisaruGuruge) [inline websubhub-ballerina/tests/additional_error_details_test.bal:117]: We can do this and remove the type guard?

Check other places as well.
```suggestion
    string payload = check response.getTextPayload();
```

**Review by kalaiyarasiganeshalingam — APPROVED** [review]

**Review by Bhashinee — APPROVED** [review]

**Pull request merged**